packages/apollo-collaboration-server/src/fileHandling/fileHandling.service.ts

```diff
@@ -503,4 +508,107 @@ export class FileHandlingService {
+    const stringOfGFF3 = await fs.readFile(filenameWithPath, {
+      encoding: 'utf8',
+      flag: 'r',
+    })
+    this.logger.verbose(`Data read from file=${stringOfGFF3}`)
+
+    const arrayOfThings = gff.parseStringSync(stringOfGFF3, {
+      parseAll: true,
+    })
+    this.logger.debug(`Setting apollo_ids if required....`)
+
+    // Loop all lines and check if each line has 'apollo_id' property inside attributes
+    for (const entry of arrayOfThings) {
+      // Comment, Directive and FASTA -entries are not presented as an array
+      if (Array.isArray(entry)) {
+        for (const [key, val] of Object.entries(entry)) {
+          this.logger.verbose(
+            `GFF3Item: key=${JSON.stringify(key)}, value=${JSON.stringify(
+              val,
+            )}`,
+          )
+          if (val.hasOwnProperty('attributes')) {
+            const assignedVal = Object.assign(val)
+            // Let's add apollo_id to parent feature if it doesn't exist
+            if (!assignedVal.attributes.hasOwnProperty('apollo_id')) {
+              assignedVal.attributes.apollo_id = uuidv4()
+            }
+            // Check if there is also childFeatures in parent feature and it's not empty
+            if (
+              val.hasOwnProperty('child_features') &&
+              Object.keys(assignedVal.child_features).length > 0
+            ) {
+              // Let's add apollo_id to each child recursively
+              this.setApolloIdRecursively(assignedVal)
+            }
+          }
+        }
+      }
+    }
+
+    // Save into file
+    const { FILE_SEARCH_FOLDER, GFF3_DEFAULT_FILENAME_AT_STARTUP } = process.env
+    if (!FILE_SEARCH_FOLDER) {
+      throw new Error('No FILE_SEARCH_FOLDER found in .env file')
+    }
+    if (!GFF3_DEFAULT_FILENAME_AT_STARTUP) {
+      throw new Error('No GFF3_DEFAULT_FILENAME_AT_STARTUP found in .env file')
+    }
+    await fs.writeFile(
+      join(FILE_SEARCH_FOLDER, GFF3_DEFAULT_FILENAME_AT_STARTUP),
+      gff.formatSync(arrayOfThings),
+    )
+  }
+
+  /**
+   * Loop child features in parent feature and add apollo_id to each child's attribute
+   * @param parentFeature - Parent feature
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setApolloIdRecursively(parentFeature: any) {
+    this.logger.verbose(
+      `Value in recursive method = ${JSON.stringify(parentFeature)}`,
+    )
+    // If there is child features and size is not 0
+    if (
+      parentFeature.hasOwnProperty('child_features') &&
+      Object.keys(parentFeature.child_features).length > 0
+    ) {
+      // Loop each child feature
+      for (
+        let i = 0;
+        i < Object.keys(parentFeature.child_features).length;
+        i++
+      ) {
+        this.logger.verbose(
+          `Child no #${i} has value=${JSON.stringify(
+            parentFeature.child_features[i][0],
+          )}`,
+        )
+        const assignedVal = Object.assign(parentFeature.child_features[i][0])
```

Comment:
This should have a type, same as above. The lines below will need to be updated, too.

```suggestion
        const assignedVal: GFF3FeatureLineWithRefs = Object.assign(
          parentFeature.child_features[i][0],
        )
```